### PR TITLE
[OIS] Add PSA crypto backend support

### PR DESCRIPTION
**Pull request by ATmobica**
These changes allow the selection of PSA as the cryptographic algorithm used when building Matter CryptoPAL with the Open IoT SDK.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

File: .github/workflows/examples-openiotsdk.yaml [hunk 1]
```diff
@@ -31,6 +31,11 @@ env:
 
 jobs:
     openiotsdk:
+        strategy:
+            fail-fast: false
+            matrix:
+                cryptoBackend: ["psa", "mbedtls"]
```
[the remark this answers is not in the document]
Comment (ATmobica) [reply .github/workflows/examples-openiotsdk.yaml:37]: I think the minimum validation set for both crypto backends is: example (can be lock-app for example) + unit test. I'm not sure if it's better to create one long-term job or two parallel ones. 
But of course, it is possible to change that if you want.

[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply .github/workflows/examples-openiotsdk.yaml:37]: I will leave the final decision to you but it's always worth limiting the number of combinations to verify so I wanted to double-check that you considered that option.

**Comment by ATmobica** [comment]
> @ATmobica this seems to have been merged by mergify however after a non-blocking CI failure, so master is permanently red.
> 
> Should we revert or can we fix OpenIOT?

Let me check if we can fix this quickly. If not we should revert the merged changes.

**Comment by woody-apple** [comment]
Sorry, need to revert this. Holding up CI :(

**Pull request merged**